spec/USERS.md

```diff
@@ -0,0 +1,70 @@
+- clusterAdmin
+
+By default, `root_user` is given a username of `root-user` and a password of `password`.
+However, you may override these defaults by specifying a username and password in the
+`MONGODB_URI` environment variable while running your tests. For example, if you set `MONGODB_URI` to: `mongodb://alanturing:enigma@localhost:27017/`, the username of `root_user` would be set to `alanturing`, and the password would be set to `enigma`.
+
+## test_user
+`test_user` is the user created with a more limited set of privileges. It is created with the following
+roles:
+- readWrite on the ruby-driver database
+- dbAdmin on the ruby-driver database
+
+It is also granted the following roles against a database called "invalid_database." These permissions are used for the purpose of running tests against a database that doesn't exist.
+- readWrite on the invalid_database database
+- dbAdmin on the invalid_database database
+
+`test_user` also has the following roles, which are exclusively used to test transactions:
+- readWrite on the hr database
+- dbAdmin on the hr database
+- readWrite on the reporting database
+- dbAdmin on the reporting database
+
+The `test_user` has the username `test-user` and the password `password`; these values are not customizable without changing the source code.
+
+## User Creation
+
+Both users are typically created in the [spec_setup](support/spec_setup.rb) script, which can be
+run in two ways: either by running `bundle exec rake spec:prepare`, which only runs spec setup without
+running any actual tests, or by running `rake`, which runs spec setup and the entire test suite.
+
+First, the `spec_setup` script attempts to create the `root_user`. If this user already exists (for example,
+if you have already created this user in your test instance), `spec_setup` will skip this step. Once
+the script has verified the existence of `root_user`, it will create a client authenticated with the `root_user` and use that client to create a second user, `test_user`. Because `root_user` has the `userAdminAnyDatabase` role, it has the permissions necessary to create and destroy users on your MongoDB instance. If you have already created a user with the same credentials as `test_user` prior to running
+the `spec_setup` script, the script will delete this user and re-create it.
+
+The `root_user` is created in the `admin` database, while the `test_user` is created in the `ruby-driver`
+database.
+
+The authentication mechanism used to store the user credentials is going to change depending on the version of MongoDB running on your deployment. If you are running tests against an instance running a MongoDB server version older than 3.2, the users are created using the `MONGODB-CR` authentication mechanism. Starting in version 3.2,`MONGODB-CR` was replaced as the default authentication mechanism, so on any test instance running MongoDB 3.2 or newer, users will be created using the `SCRAM` authentication mechanism by default.
```

Comment:
There are two scram mechanisms, right? "SCRAM mechanism" isn't actually a thing?

Reply:
You're right! I'll adjust.